From b5b9fd01c4db135893c44e82a9f33c2411e993d0 Mon Sep 17 00:00:00 2001
From: Takayuki 'January June' Suwa <jjsuwa_sys3175@yahoo.co.jp>
Date: Fri, 6 May 2022 19:34:06 +0900
Subject: [PATCH 03/31] xtensa: Rename deprecated extv/extzv insn patterns to
 extvsi/extzvsi

These patterns were deprecated since GCC 4.8.

gcc/ChangeLog:

	* config/xtensa/xtensa.md (extvsi, extvsi_internal, extzvsi,
	extzvsi_internal): Rename from extv, extv_internal, extzv and
	extzv_internal, respectively.
---
 gcc/config/xtensa/xtensa.md | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/gcc/config/xtensa/xtensa.md b/gcc/config/xtensa/xtensa.md
index 123916957..251c313d5 100644
--- a/gcc/config/xtensa/xtensa.md
+++ b/gcc/config/xtensa/xtensa.md
@@ -634,7 +634,7 @@
 
 ;; Field extract instructions.
 
-(define_expand "extv"
+(define_expand "extvsi"
   [(set (match_operand:SI 0 "register_operand" "")
 	(sign_extract:SI (match_operand:SI 1 "register_operand" "")
 			 (match_operand:SI 2 "const_int_operand" "")
@@ -649,12 +649,12 @@
   if (!lsbitnum_operand (operands[3], SImode))
     FAIL;
 
-  emit_insn (gen_extv_internal (operands[0], operands[1],
-				operands[2], operands[3]));
+  emit_insn (gen_extvsi_internal (operands[0], operands[1],
+				  operands[2], operands[3]));
   DONE;
 })
 
-(define_insn "extv_internal"
+(define_insn "extvsi_internal"
   [(set (match_operand:SI 0 "register_operand" "=a")
 	(sign_extract:SI (match_operand:SI 1 "register_operand" "r")
 			 (match_operand:SI 2 "sext_fldsz_operand" "i")
@@ -669,7 +669,7 @@
    (set_attr "mode"	"SI")
    (set_attr "length"	"3")])
 
-(define_expand "extzv"
+(define_expand "extzvsi"
   [(set (match_operand:SI 0 "register_operand" "")
 	(zero_extract:SI (match_operand:SI 1 "register_operand" "")
 			 (match_operand:SI 2 "const_int_operand" "")
@@ -678,12 +678,12 @@
 {
   if (!extui_fldsz_operand (operands[2], SImode))
     FAIL;
-  emit_insn (gen_extzv_internal (operands[0], operands[1],
-				 operands[2], operands[3]));
+  emit_insn (gen_extzvsi_internal (operands[0], operands[1],
+				   operands[2], operands[3]));
   DONE;
 })
 
-(define_insn "extzv_internal"
+(define_insn "extzvsi_internal"
   [(set (match_operand:SI 0 "register_operand" "=a")
 	(zero_extract:SI (match_operand:SI 1 "register_operand" "r")
 			 (match_operand:SI 2 "extui_fldsz_operand" "i")
-- 
2.20.1

